pennylane/measurements/shots.py

```diff
@@ -279,3 +279,6 @@ def bins(self):
                 upper_bound = lower_bound + sc.shots
                 yield lower_bound, upper_bound
                 lower_bound = upper_bound
+
+
+ShotsLike = Union[Shots, int, Sequence[Union[int, tuple[int, int]]]]
```

Comment:
```suggestion
ShotsLike = Union[Shots, None, int, Sequence[Union[int, tuple[int, int]]]]
```
`shots=None` always is acceptable, as it means "analytic infinite shots".

Reply:
The argument is specified as `shots: Optional[ShotsLike] = None`, which already covers the `None` case.